tools/backups2datalad.py

```diff
@@ -422,17 +421,29 @@ def get_annex_hash(filepath):
                         # ATM. TODO: discussion is ongoing to change API
                         # see e.g. https://github.com/dandi/dandi-api/pull/382
                         blob_id_ret = a.client.post(
-                                '/blobs/digest/', 
-                                json={'algorithm': 'dandi:dandi-etag', 
-                                      'value': new_metadata.digest[models.DigestType.dandi_etag]})
+                            "/blobs/digest/",
+                            json={
+                                "algorithm": "dandi:dandi-etag",
+                                "value": new_metadata.digest[
+                                    models.DigestType.dandi_etag
+                                ],
+                            },
+                        )
                         # no such API interface yet!
                         # a.set_raw_metadata(new_metadata)
-                        ret = a.client.put(a.api_path, 
-                                json={'metadata': new_metadata_json, 
-                                      'blob_id': blob_id_ret['blob_id']})
-                        assert adict['asset_id'] != ret['asset_id']
-                        assert adict['path'] == ret['path']
-                        adict = ret
+                        ret = a.client.put(
+                            a.api_path,
+                            json={
+                                "metadata": new_metadata_json,
+                                "blob_id": blob_id_ret["blob_id"],
+                            },
+                        )
+                        assert adict["asset_id"] != ret["asset_id"]
+                        assert adict["path"] == ret["path"]
+                        # Update adict in-place so that the instance appended
+                        # to asset_metadata gets updated:
+                        adict.clear()
+                        adict.update(ret)
```

Comment:
oh, may be that is what contributed to the yesterday's dance with delete/re-add on 000055 . thanks!

I will merge, but I think @satra has prepared an extract of this to be ran independently now.